### PR TITLE
renderer: add texture dumping

### PR DESCRIPTION
**Pull request by sunho**
# About PR

Based on https://github.com/Vita3K/Vita3K/pull/522 by @VelocityRa 

I elaborated logging and supported compressed texture dumping.

### CODE REVIEW

File: vita3k/renderer/src/gl/texture.cpp [hunk 1]
```diff
@@ -283,5 +285,67 @@ void upload_bound_texture(const SceGxmTexture &gxm_texture, const MemState &mem)
     }
 }
 
+// Dumps bound texture to a file
+void dump(const SceGxmTexture &gxm_texture, const MemState &mem, const std::string &parameter_name, const std::string &base_path, const std::string &title_id, Sha256Hash program_hash) {
+    static uint32_t gg_tex_index = 0;
```
Comment (VelocityRa) [inline vita3k/renderer/src/gl/texture.cpp:290]: you meant `g_tex_index`?
`g_` is supposed to be for globals and below you use it for a local instead
not sure what "gg" is

Comment (sunho) [reply vita3k/renderer/src/gl/texture.cpp:290]: Absolutely right. I fixed it.

File: vita3k/renderer/src/state_set.cpp [hunk 2]
```diff
@@ -312,12 +312,15 @@ COMMAND_SET_STATE(stencil_ref) {
 COMMAND_SET_STATE(fragment_texture) {
     const std::uint32_t texture_index = helper.pop<std::uint32_t>();
     SceGxmTexture texture = helper.pop<SceGxmTexture>();
-
+    auto frag_program = state->fragment_program.get(mem);
+    auto program = frag_program->program.get(mem);
+    const Sha256Hash hash_bytes = sha256(program, program->size);
```
Comment (VelocityRa) [inline vita3k/renderer/src/state_set.cpp:317]: isn't this is hashing every texture even if texture dumping isn't enabled?
there should be no additional hashing if it's disabled

File: vita3k/renderer/src/gl/sync_state.cpp [hunk 3]
```diff
@@ -327,6 +344,8 @@ bool sync_state(GLContext &context, const GxmContextState &state, const MemState
 
     // Textures.
     const SceGxmFragmentProgram &gxm_fragment_program = *state.fragment_program.get(mem);
+    const auto program = gxm_fragment_program.program.get(mem);
+    const Sha256Hash hash_bytes = sha256(program, program->size);
```
Comment (VelocityRa) [inline vita3k/renderer/src/gl/sync_state.cpp:348]: same here

File: vita3k/renderer/src/state_set.cpp (hunk 2 above)
Comment (sunho) [reply vita3k/renderer/src/state_set.cpp]: fixed!

File: vita3k/renderer/src/gl/sync_state.cpp (hunk 3 above)
Comment (sunho) [reply vita3k/renderer/src/gl/sync_state.cpp]: fixed!

**Pull request merged**